backstory/src/main/java/com/google/sps/servlets/AnalyzeImageServlet.java

```diff
@@ -0,0 +1,311 @@
+ */
+@WebServlet("/analyze-image")
+public class AnalyzeImageServlet extends HttpServlet {
+  private ImagesManager imagesManager;
+  private Boolean useMockStoryManager;
+  private StoryManager storyManager;
+  private StoryAnalysisManager storyAnalysisManager;
+  private DatastoreService datastore;
+  private Boolean useMockDatastoreService;
+  private BlobstoreService blobstoreService;
+  private Boolean useMockBlobstoreService;
+  private Entity mockAnalyzedImageEntity;
+
+  public AnalyzeImageServlet() throws IOException, APINotAvailableException {
+    imagesManager = new VisionImagesManager();
+    useMockStoryManager = false;
+    storyAnalysisManager = new PerspectiveStoryAnalysisManager();
+    useMockDatastoreService = false;
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    useMockBlobstoreService = false;
+    blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+  }
+
+  // Injection methods for testing.
+
+  /**
+   * Sets an optional images manager.
+   * @param imagesManager an ImagesManager object, real or mock.
+   */
+  public void setToUseMockImagesManager(ImagesManager imagesManager) {
+    this.imagesManager = imagesManager;
+  }
+
+  /**
+   * Sets an optional storyAnalysisManager.
+   * @param storyAnalysisManager an StoryAnalysisManager object, real or mock.
+   */
+  public void setToUseMockStoryAnalysisManager(StoryAnalysisManager storyAnalysisManager) {
+    this.storyAnalysisManager = storyAnalysisManager;
+  }
+
+  /**
+   * Sets optional Datastore services and entities.
+   * @param datastore a datastore service object, real or mock.
+   * @param mockAnalyzedImageEntity an entity object, real or mock.
+   */
+  public void setToUseMockDatastoreService(
+      DatastoreService datastore, Entity mockAnalyzedImageEntity) {
+    useMockDatastoreService = true;
+    this.datastore = datastore;
+    this.mockAnalyzedImageEntity = mockAnalyzedImageEntity;
+  }
+
+  /**
+   * Sets optional Blobstore services.
+   * @param blobstoreService a Blobstore service object, real or mock.
+   */
+  public void setToUseMockBlobstoreService(BlobstoreService blobstoreService) {
+    useMockBlobstoreService = true;
+    this.blobstoreService = blobstoreService;
+  }
+
+  /**
+   * Sets an optional StoryManager.
+   * @param storyManager a StoryManager object, real or mock.
+   */
+  public void setToUseMockStoryManager(StoryManager storyManager) {
+    useMockStoryManager = true;
+    this.storyManager = storyManager;
+  }
+
+  /**
+   * Helper method to set a real story manager object, if a mock is not being used.
+   * @param prompt the small prompt used to generate a larger story.
+   * @param storyLength number of words to be in the backstory.
+   * @param temperature 0-1 paramater which determines the freedom of expression given to
+   * the text generation model.
+   */
+  private void createStoryManager(String prompt, int storyLength, double temperature) {
+    if (!useMockStoryManager) {
```

Comment:
It's discouraged to invoke test code from prod code... it increases the risk that test behavior will ship.